koku/api/iam/test/tests_user_preference_view.py

```diff
@@ -0,0 +1,310 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Test the IAM views."""
+
+from random import randint
+
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from .iam_test_case import IamTestCase
+from ..models import Customer, User, UserPreference
+
+
+class UserPreferenceViewtest(IamTestCase):
+    """Tests the UserPreference view."""
+
+    def setUp(self):
+        """Set up the user view tests."""
+        super().setUp()
+        self.create_service_admin()
+
+        self.test_data = {'name': 'test-pref',
+                          'description': 'test-pref',
+                          'preference': {'test-pref': ['a',
```

Comment:
I think we should put controls around what we want to allow to be stored in preference. Don't want to allow storing random JSON data but the intended preferences called out in https://github.com/project-koku/koku/issues/97.